### ISSUE TITLE
After update to 1.8 version and when using the ASP .Net Core Prometheus exporter and /metrics end point I get error on Prometheus some times. 

### ISSUE THREAD

**Issue by jnormen** [open-telemetry/opentelemetry-dotnet#5506]
# Bug Report

List of [all OpenTelemetry NuGet
packages](https://www.nuget.org/profiles/OpenTelemetry) and version that you are
using (e.g. `OpenTelemetry 1.0.2`):

 <PackageVersion Include="OpenTelemetry" Version="1.8.0" />
 <PackageVersion Include="OpenTelemetry.Api" Version="1.5.1" />
 <PackageVersion Include="OpenTelemetry.Exporter.Console" Version="1.8.0" />
 <PackageVersion Include="OpenTelemetry.Exporter.Prometheus.AspNetCore" Version="1.8.0-rc.1" />
 <PackageVersion Include="OpenTelemetry.Exporter.Zipkin" Version="1.4.0" />
 <PackageVersion Include="OpenTelemetry.Extensions.Hosting" Version="1.8.0" />
 <PackageVersion Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.7.1" />
 <PackageVersion Include="OpenTelemetry.Instrumentation.Http" Version="1.0.0-rc9.8" />
 <PackageVersion Include="OpenTelemetry.Exporter.OpenTelemetryProtocol" Version="1.8.0" />
 <PackageVersion Include="OpenTelemetry.Exporter.OpenTelemetryProtocol.Logs" Version="1.4.0-beta.2" />
 
 Metrics I use:
 .AddMeter("Microsoft.AspNetCore.Hosting",
          "Microsoft.AspNetCore.Routing",
          "Microsoft.AspNetCore.Diagnostics",
          "Microsoft.AspNetCore.Server.Kestrel",
          "Polly",
          "MassTransit"
 
## Symptom

PrometheusTargetMissing *****   (http monitor/prometheus critical)[FIRING:3]

With info:
A Prometheus target has disappeared. An exporter might be crashed.

The text mention: 
"expected equal, got INVALID" 

When I revert to version 1.7 the problem stops with the same meters on. 

Same error in beta and rc1 version of 1.8 exporter. Not a valid output. 

What did you see instead?

I get alarm from Prometheus with:
description = A Prometheus target has disappeared. An exporter might be crashed. VALUE = 0 LABELS =

## Reproduce

Use 1.8 version of Open Telemetry and the 1.8 rc1 of the Prometheus asp.net core exporter

**Comment by Abrynos** [comment]
This issue is caused by Prometheus adding the `Accept: application/openmetrics-text` header. I've got this reproducible with both Prometheus version `2.45.3+ds-3` (currently in Debian testing) and with 
```bash
curl -H "Accept: application/openmetrics-text" {protocol}://{host}:{port}/{configured_endpoint} # incorrect response body
```


In comparison, the response comes out flawlessly using either command-line below or `OpenTelemetry.Exporter.Prometheus.AspNetCore` version `1.7.0-rc1` or earlier
```bash
curl {protocol}://{host}:{port}/{configured_endpoint} # works
```

Here's a sample response camptured via `curl`: [test.txt](https://github.com/open-telemetry/opentelemetry-dotnet/files/14897999/test.txt)


---

Related PR: JustArchiNet/ArchiSteamFarm#3183

Additional thanks to @ezhevita for investing your time as well.

**Comment by martinjt** [comment]
I believe this the same as #5515, which is being fixed. If that seems right, we can close this in favour of that.

**Comment by reyang** [comment]
Duplicated with #5515. The bug was introduced by #5107 and fixed by #5623.
